contrib/babelfishpg_tds/src/backend/tds/tdssecure.c

```diff
@@ -245,6 +274,28 @@ SslHandShakeWrite(BIO * h, const char *buf, int size)
 	return (res - TDS_PACKET_HEADER_SIZE);
 }
 
+
+ssize_t
+test_ssl_handshake_write(BIO * h, char *buf, int size, TdsSecureSocketApiConst mock_socket_write)
+{
+
+    /*
+     *  Accessing SslHandShakeWrite() function directly from our testing functions created in the test_ssl_write.c file is not possible due to its static nature
+     *  This intermediary function serves as a gateway, allowing us to invoke the SslHandShakeWrite() function from our testing function
+     */
+
+    int res;
+    tds_secure_raw_write = mock_socket_write;
+    unit_testing = true;
```

Comment:
what if there is an error during testing and unit_testing remains true? I think not an issue in current set of tests but could be a problem later

Reply:
Handles this using PG_TRY() and PG_FINALLY() block.